crates/hir-def/src/nameres/collector.rs

```diff
@@ -1112,6 +1111,84 @@ impl DefCollector<'_> {
         }
     }
 
+    fn push_res_and_update_glob_vis(
+        &mut self,
+        module_id: LocalModuleId,
+        name: &Name,
+        mut defs: PerNs,
+        def_import_type: Option<ImportType>,
+    ) -> bool {
+        let mut changed = false;
+
+        if let Some(ImportType::Glob(_)) = def_import_type {
+            let prev_defs = self.def_map[module_id].scope.get(name);
+
+            // Multiple globs may import the same item and they may override visibility from
+            // previously resolved globs. Handle overrides here and leave the rest to
+            // `ItemScope::push_res_with_import()`.
+            if let Some((def, def_vis, _)) = defs.types {
+                if let Some((prev_def, prev_vis, _)) = prev_defs.types {
```

Comment:
I believe the third parameter here is the location/token the import occurred at - it seems it wasn't present at the time the code was written, but I don't think there's a condition necessary on it here, so hopefully it's correct to leave it ignored here.

If you can think of any test cases to add to validate if there are other necessary conditions here, I'd appreciate any suggestions!